Signup login homepage week8

simaoalicechen opened the pull request:
This PR includes: 

1. Sign-up (fully functioning with a fraction of Userfront API functions);
2. Log-in (without Userfront, since I am not sure if it is a good idea to connect with the backend using userfront);

Have a look. Please do not modify given the above circumstances. We can talk after standup.

simaoalicechen committed:
Basic signup container and fields working

simaoalicechen committed:
Added single sign on with Google

simaoalicechen committed:
Added all login and signup layout code

Most of the layout is done. The functionality is almost done, a few bugs exist in alerts and setting password and email rules. These logics will be done tomorrow.

simaoalicechen committed:
Layout and basic function working

Some fine tuning could include adding Userfront API to implement the login and signup function. Other things to consider include nav bar.

simaoalicechen committed:
Added Nav bar

simaoalicechen committed:
Added Nav bar. All layout complete

simaoalicechen committed:
Merge branch 'develop' into Signup_Login_Homepage_Week8

simaoalicechen commented:
Added Nav bar today. 

Things to discuss: 

1. Are we going to use Userfront? Will it be compatible for the backend data/json file setup? Either way is possible: to keep using it or take it out and write our own logics for check password and email verifiability. 
2. Linter Javascript PR Travis is failing. Please advise.

simaoalicechen commented:
@Pooja-Codes I can't find the json file written by Daniel that supposedly has username and password. Do you mind let me know the directory?

Pooja-Codes commented:
> @Pooja-Codes I can't find the json file written by Daniel that supposedly has username and password. Do you mind let me know the directory?

There is no JSON file! I never said that Daniel wrote json file. We (you or me) have to create that for our own reference which will include only "username" and "password" for each user. It is easy but depends on your implementation of the sign-up page.

We are making it simple so that Alex can set up the initial model quickly and then later we can add more fields. By saying that, I would also like to share that Alex doesn't require anything from you.

Pooja-Codes commented:
> Added Nav bar today.
> 
> Things to discuss:
> 
> 1. Are we going to use Userfront? Will it be compatible for the backend data/json file setup? Either way is possible: to keep using it or take it out and write our own logics for check password and email verifiability.
> 2. Linter Javascript PR Travis is failing. Please advise.

For Linter JS Fail fix, I would suggest going to the details of it and following the suggested fix steps by the Linter itself.

simaoalicechen committed:
Simplified login and signup codes, and varaible names, and edited Nav Bar so that it is more pleasing

All logics are good. One thing to remember is that we don't have to check the signin email format, because if it is not correct, they can't sign in. In addition, currently, the login error message is that if you don't enter password, it reads the password is not matching, but this is just to check if the error message works. What really checks that is after backend data can save the password. Finally, I also simplified certain varaible names so that it would be more readable. I have not been able to connect this with the API json file, since I can't find it in the directory. However, it should be easy for backend engineers to read this file, as it is very clear now. I can also do a quick walkthrough, if anyone has questions about this frontend codes.

simaoalicechen commented:
> > @Pooja-Codes I can't find the json file written by Daniel that supposedly has username and password. Do you mind let me know the directory?
> 
> There is no JSON file! I never said that Daniel wrote json file. We (you or me) have to create that for our own reference which will include only "username" and "password" for each user. It is easy but depends on your implementation of the sign-up page.
> 
> We are making it simple so that Alex can set up the initial model quickly and then later we can add more fields. By saying that, I would also like to share that Alex doesn't require anything from you.

Aw, this makes sense now. Okay. Do you want to talk about this either before/after the standup meeting tomorrow?

simaoalicechen commented:
> > Added Nav bar today.
> > Things to discuss:
> > 
> > 1. Are we going to use Userfront? Will it be compatible for the backend data/json file setup? Either way is possible: to keep using it or take it out and write our own logics for check password and email verifiability.
> > 2. Linter Javascript PR Travis is failing. Please advise.
> 
> For Linter JS Fail fix, I would suggest going to the details of it and following the suggested fix steps by the Linter itself.

I find those linter details and messages now. I will fix them tomorrow. I will not be available after this message, but please feel free to set up a meeting to meet sometime tomorrow. (Preferably after the standup meeting). Thanks.

simaoalicechen committed:
Added eslint and see if it will pass travis

simaoalicechen committed:
Seperated all components in homepage

Next step is to add switch function so that the users can choose if they want to sign up or log in

simaoalicechen committed:
Deleted some redundant css

simaoalicechen committed:
Added Sign up and Log in Siwtch function so that we have seperate views

Note on how to approach the files: All most updated working files are in switchViews directory, and they are linked properly to things in DemoView, LoginView, and SignupView in views folder.

simaoalicechen committed:
Changed Navbar fonts

simaoalicechen committed:
Merge branch 'develop' into Signup_Login_Homepage_Week8

simaoalicechen committed:
checking linter

simaoalicechen committed:
Solved a bunch of conflicts and linter problems

simaoalicechen committed:
Linter solved

simaoalicechen committed:
Linter works locally

simaoalicechen committed:
Delete all userfront and retried linter

simaoalicechen committed:
Linter solved and deleted redundant files

Pooja-Codes commented on src/frontend/shortlist/.eslintrc.cjs:
```diff
@@ -3,6 +3,9 @@ require("@rushstack/eslint-patch/modern-module-resolution");
 
```
I am unsure about this file. Have you created this or it is automatically by linter? Also, Can you explain what changes you have done as part of fixing the Linter failure?

simaoalicechen replied:
This was automatically created when I installed eslinter. It did not work initially. I added the dictionary-style commands from a previous version of an .eslintrc.js file that worked for that version. Then it works here too. The effective lines are the ones I added here.

simaoalicechen replied:
Specifically, I think these two lines solved the problems: 
1. parser: "@babel/eslint-parser", (prevents parsing issues)
2. rules: {
    "vue/multi-word-component-names": 0,
  },  (allows your personalized naming conventions)

Pooja-Codes commented on src/frontend/shortlist/.eslintrc.js:
```diff
@@ -0,0 +1,24 @@
+/* eslint-env node */
```
I am unsure about this file. Have you created this or it is automatically by linter? Also, Can you explain what changes you have done as part of fixing the Linter failure?

simaoalicechen replied:
This file is created automatically when I installed linter locally. I don't think it cause any issue when merging into develop, because it is a well-established file that is meant to set some rules. I do have some caution around it, but I think it is just like a README.MD file that you created whenever you start a new repo. Does no harm and helps my branch work better. I highly doubt it would cause any trouble to other people's codes.

Pooja-Codes commented on src/frontend/shortlist/cypress.config.js:
```diff
@@ -1,8 +1,8 @@
-    specPattern: "cypress/e2e/**/*.{cy,spec}.{js,jsx,ts,tsx}",
-    baseUrl: "http://localhost:4173",
-  },
+    e2e: {
```
I am unsure about this file. Have you created this? Also, it looks like there are no changes.

simaoalicechen replied:
I have no idea either. last time I checked, GitHub told me AZ added these lines in git log. You can check yourself too.

Pooja-Codes commented on src/frontend/shortlist/package.json:
```diff
@@ -8,7 +8,8 @@
     "test:unit": "vitest --environment jsdom",
```
I am unsure about this file. Have you created this or it is automatically by linter? Also, Can you explain what changes you have done as part of fixing the Linter failure?

simaoalicechen replied:
I never changed anything of this file. I don't think this line would cause any harm to any file either.

Pooja-Codes commented on src/frontend/shortlist/src/components/layout/signup_login/Logo.vue:
```diff
@@ -0,0 +1,27 @@
+<script>
+export default {
```
I guess the purpose of separating this file is so that we can add a features bar below the logo. Correct me if I am wrong.

simaoalicechen replied:
Yes, you are correct.

Pooja-Codes commented on src/frontend/shortlist/src/components/layout/signup_login/Loginview.vue:
```diff
@@ -0,0 +1,126 @@
+      // For log in
+      username_login: "",
+      password_login: "",
+      email_login: "",
```
Do we require an email address for login? as we verify email in the signup process and then the user's account gets created with username and password.

simaoalicechen replied:
Yes.... by user-friendly convention. I understand what you said about no mentioning of email in js file to make things easier for Alex. But it would be comical if we don't require email for signup.

Pooja-Codes commented on src/frontend/shortlist/src/views/LoginView.vue:
```diff
@@ -0,0 +1,14 @@
+<script>
```
It is a redundant file as there is a Demo file that can handle login and signup both. Should we remove it?

simaoalicechen replied:
This is not a redundant file. Do not remove. I may change the name later.

Pooja-Codes commented on src/frontend/shortlist/src/views/SignupView.vue:
```diff
@@ -0,0 +1,14 @@
+<script>
```
It is a redundant file as there is a Demo file that can handle login and signup both. Should we remove it?

simaoalicechen replied:
See comment above.

Pooja-Codes commented on src/frontend/shortlist/src/components/layout/signup_login/Signupview.vue:
```diff
@@ -0,0 +1,127 @@
+<script>
```
I would suggest keeping the file names without "view" and "View" so that later we don't have to work on renaming and organizing things. For example, keep Login.vue and avoid LoginView.vue or Loginview.vue.

simaoalicechen replied:
This is a good comment. Yes, I will change the name.

Pooja-Codes commented on src/frontend/shortlist/src/views/PreferencesView.vue:
```diff
@@ -0,0 +1,14 @@
+<script>
```
Unsure about this file's presence here.

simaoalicechen replied:
I added it so that you and Daniel's previous work on preferences can be shown. It needs to be a url regardless, and in this case, it should be linked to the home page too.

Pooja-Codes reviewed (changes requested):
It would be great if you can look over some changes regarding file names, location, and minor variable presence on the login page. 

Nice work!

simaoalicechen committed:
Changed the file names in signup_login directory to reduce confusion

simaoalicechen committed:
Used the old cjs and kept the new js for eslint.

simaoalicechen committed:
Deleted eslintrc.cjs

simaoalicechen committed:
Deleted cjs file and changed the json files

simaoalicechen committed:
Changed json file and eslint file accordingly

Pooja-Codes reviewed (approved):
LGTM! Awesome Work!

simaoalicechen committed:
Added test
